lib/utils/processOptions.js

```diff
@@ -23,7 +24,21 @@ function processOptions(config, argv, callback) {
   // we should use portfinder.
   const options = createConfig(config, argv, { port: defaultPort });
 
-  callback(config, options);
+  if (options.socket) {
+    callback(config, options);
+  } else {
+    findPort(options.port)
+      .then((port) => {
+        options.port = port;
+        callback(config, options);
+      })
+      .catch((err) => {
+        // eslint-disable-next-line no-console
+        console.error(err.stack || err);
+        // eslint-disable-next-line no-process-exit
+        process.exit(1);
```

Comment:
I think it should be `throw err;` here as it was before

Reply:
Doing so would cause an unhandled promise rejection, which might go unnoticed by the user of webpack-dev-server. Right? Why is a rejected promise better than exiting the process?

I chose to follow the pattern [here](https://github.com/webpack/webpack-dev-server/pull/2143/files#diff-08aa20d4f0b23d049d0ea9d0377a03d8R16) since it seemed more likely to be helpful to the user. Am very happy to be corrected, though, if I'm missing something :)

Reply:
To add to this, my understanding is that `.catch(err => {throw err})` is essentially a no-op that just makes an already rejected promise remain rejected with the same err. So I went looking for something better and found the process.exit() code from a few lines above my code.

Reply:
You are right, this looks better